### PR TITLE
ddt removal in ui location tests

### PR DESCRIPTION
**Pull request by sthirugn**
```sh
# nosetests tests/foreman/ui/test_location.py 
......................E....
======================================================================
ERROR: @test: Remove environment by using location name & environment name
----------------------------------------------------------------------
nailgun.client: WARNING: Received HTTP 422 response: {
  "error": {"id":null,"errors":{"name":["is alphanumeric and cannot contain spaces"]},"full_messages":["Name is alphanumeric and cannot contain spaces"]}
}
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 27 tests in 3637.678s

FAILED (errors=1)
```

### CODE REVIEW

**Comment by sthirugn** [comment]
Failed test is fixed and reran:
```sh
# nosetests tests/foreman/ui/test_location.py:Location.test_remove_environment
.
----------------------------------------------------------------------
Ran 1 test in 82.253s

OK
```

File: tests/foreman/ui/test_location.py [hunk 1]
```diff
@@ -168,12 +184,15 @@ def test_positive_update_with_different_names(self, new_name):
         @assert: Location name is updated
 
         """
-        loc_name = gen_string('alpha')
         with Session(self.browser) as session:
+            loc_name = gen_string('alpha')
```
Comment (omaciel) [inline tests/foreman/ui/test_location.py]: Move it outside the `Session` as before (we discussed this today and agreed to do it)

**Comment by omaciel** [comment]
ACK

File: tests/foreman/ui/test_location.py [hunk 2]
```diff
@@ -770,34 +800,38 @@ def test_remove_configtemplate(self, testdata):
...
-            # Item is listed in 'Selected Items' list and not 'All Items' list.
-            self.assertIsNotNone(element)
-            self.template.delete(template)
-            self.location.search(loc_name).click()
-            session.nav.click(tab_locators['context.tab_template'])
-            element = session.nav.wait_until_element(
-                (strategy, value % template))
-            # Item is listed in 'All Items' list and not 'Selected Items' list.
-            self.assertIsNone(element)
+            for testdata in valid_config_template_names():
```
Comment (oshtaier) [inline tests/foreman/ui/test_location.py]: It seems that BZ 1164247 was closed a long time ago and is not actual anymore, so I propose to remove `valid_config_template_names()` method and that BZ part:
```
bug_id = testdata.pop('bugzilla', None)
if bug_id is not None and bz_bug_is_open(bug_id):
    self.skipTest(
          'Bugzilla bug {0} is open.'.format(bug_id))
```

**Comment by oshtaier** [comment]
There are also bunch of code mistakes or mistakes in logic. For example, when I see:
```
element = session.nav.wait_until_element(
    (strategy, value % template))
# Item is listed in 'Selected Items' list and not
# 'All Items' list.
element = session.nav.wait_until_element(
    (strategy, value % template))
# Item is listed in 'All Items' list and not
# 'Selected Items' list.
self.assertIsNone(element)
```
I want to say 'What the hell?' :)
But that is definitely out of scope current PR

**Comment by oshtaier** [comment]
Pending 1 comment

File: tests/foreman/ui/test_location.py (hunk 2 above)
Comment (sthirugn) [inline tests/foreman/ui/test_location.py]: Good idea. :+1:

**Comment by sthirugn** [comment]
@oshtaier 
`I want to say 'What the hell?' :) Why does update method remove something? And etc
But that is definitely out of scope current PR`
The code example which you pasted here updates the location to move the selected entity from Selected Items to All Items and vice versa.
Regardless @omaciel and I spoke about this module and this is very old and needs some real refactoring :)

**Comment by sthirugn** [comment]
comments updated and rebased:

```sh
# nosetests tests/foreman/ui/test_location.py:Location.test_remove_configtemplate
.
----------------------------------------------------------------------
Ran 1 test in 283.747s

OK
```

**Comment by oshtaier** [comment]
ACK

**Comment by oshtaier** [comment]
Yeah :) Need to look on it in future as sometimes you get confused from all these operations

**Comment by sthirugn** [comment]
Part of https://github.com/SatelliteQE/robottelo/issues/2769

**Pull request merged**